### PR TITLE
mimalloc: add version 1.7.9, 2.0.9

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **mimalloc/1.7.9,2.0.9**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mimalloc/all/conanfile.py [hunk 1]
```diff
@@ -111,13 +112,11 @@ def validate(self):
...
         if self.info.settings.compiler.cppstd:
             check_min_cppstd(self, self._min_cppstd)
-        check_min_vs(self, 191)
-        if not is_msvc(self):
-            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
-            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration(
-                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
-                )
+        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
```
Comment (prince-chrismc) [inline recipes/mimalloc/all/conanfile.py:115]: This pr needs some attention it's been so long

```suggestion
        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
```

File: recipes/mimalloc/all/conanfile.py [hunk 2]
```diff
@@ -111,13 +112,11 @@ def validate(self):
...
             check_min_cppstd(self, self._min_cppstd)
-        check_min_vs(self, 191)
-        if not is_msvc(self):
-            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
-            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration(
-                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
-                )
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
```
Comment (prince-chrismc) [inline recipes/mimalloc/all/conanfile.py:116]:
```suggestion
        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
```

File: recipes/mimalloc/all/conanfile.py [hunk 3]
```diff
@@ -144,6 +143,13 @@ def build(self):
             replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
                             "mimalloc-redirect.lib",
                             "mimalloc-redirect32.lib")
+        if self.version in ["1.7.9", "2.0.9"]:
+            replace_in_file(self, os.path.join(self.source_folder, "include", "mimalloc.h"),
```
Comment (prince-chrismc) [inline recipes/mimalloc/all/conanfile.py:147]: Can this be a patch please?

File: recipes/mimalloc/all/conanfile.py [hunk 4]
```diff
@@ -144,6 +143,13 @@ def build(self):
...
+                "template<class U> bool is_equal(const _mi_heap_stl_allocator_common<U, destroy>& x) const { return (this->heap == x.heap); }",
+                "template<class U, bool b> bool is_equal(const _mi_heap_stl_allocator_common<U, b>& x) const { return (this->heap == x.heap); }")
+            replace_in_file(self, os.path.join(self.source_folder, "include", "mimalloc.h"),
```
Comment (prince-chrismc) [inline recipes/mimalloc/all/conanfile.py:150]: Can this be a patch please?

File: recipes/mimalloc/all/conanfile.py (hunk 3 above)
Comment (toge) [reply recipes/mimalloc/all/conanfile.py:147]: @prince-chrismc 
I agree with you!
I create patches for it.

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**